tgui/packages/tgui/interfaces/ClockworkSlab.js

```diff
@@ -0,0 +1,272 @@
+              <Table.Cell>{script.desc}</Table.Cell>
+              <Table.Cell collapsing textAlign="right">
+                <Button
+                  fluid
+                  content={'Quickbind'}
+                  disabled={!script.purchased}
+                  onClick={() =>
+                    act('quickbind', {
+                      scriptureName: script.name,
+                    })
+                  }
+                />
+              </Table.Cell>
+            </TableRow>
+            <Table.Cell>
+              <Divider />
+            </Table.Cell>
+          </Fragment>
+        ) : (
+          <Box key={script} />
+        )
+      )}
+    </Table>
+  );
+};
+
+export const ClockworkOverview = (props, context) => {
+  const { data } = useBackend(context);
+  const { power, cogs, vitality, max_power, max_vitality } = data;
+  return (
+    <Box>
+      <Box color="good" bold fontSize="16px">
+        {'Celestial Gateway Report'}
+      </Box>
+      <Divider />
+      <ClockworkOverviewStat
+        title="Cogs"
+        amount={cogs}
+        maxAmount={10}
+        iconName="cog"
+        unit=""
+      />
+      <ClockworkOverviewStat
+        title="Power"
+        amount={power}
+        maxAmount={max_power}
+        iconName="battery-half "
+        overrideText={convertPower(power)}
+      />
+      <ClockworkOverviewStat
+        title="Vitality"
+        amount={vitality}
+        maxAmount={max_vitality}
+        iconName="tint"
+        unit="u"
+      />
+    </Box>
+  );
+};
+
+export const ClockworkOverviewStat = (props, context) => {
+  const { title, iconName, amount, maxAmount, unit, overrideText } = props;
+  return (
+    <Box height="22px" fontSize="16px">
+      <Grid>
+        <Grid.Column>
+          <Icon name={iconName} rotation={0} spin={0} />
+        </Grid.Column>
+        <Grid.Column size="2">{title}</Grid.Column>
+        <Grid.Column size="8">
+          <ProgressBar
+            value={amount}
+            minValue={0}
+            maxValue={maxAmount}
+            ranges={{
+              good: [maxAmount / 2, Infinity],
+              average: [maxAmount / 4, maxAmount / 2],
+              bad: [-Infinity, maxAmount / 4],
+            }}>
+            {overrideText ? overrideText : amount + ' ' + unit}
+          </ProgressBar>
+        </Grid.Column>
+      </Grid>
+    </Box>
+  );
+};
+
+export const ClockworkButtonSelection = (props, context) => {
+  const [selectedTab, setSelectedTab] = useLocalState(
+    context,
+    'selectedTab',
+    {}
+  );
+  const tabs = ['Servitude', 'Preservation', 'Structures'];
+  return (
+    <Table>
+      <Table.Row>
+        {tabs.map((tab) => (
+          <Table.Cell key={tab} collapsing>
+            <Button
+              key={tab}
+              fluid
+              content={tab}
+              onClick={() => setSelectedTab(tab)}
+            />
+          </Table.Cell>
+        ))}
+      </Table.Row>
+    </Table>
+  );
+};
```

Comment:
Don't export anything you don't expect to need outside of this file.

Reply:
Done